### ISSUE TITLE
Improve sizing of headings

### ISSUE THREAD

**Issue by waldyrious** [waldyrious/downstyler#30]
The default heading sizes are not quite appropriate -- h6 is smaller than body text, which doesn't make much sense IMO for a heading.

Since h1 appears to be consistently sized at 2em (200%) by browsers' native stylesheets, it would be nice to have a progression from h6 at the minimum acceptable heading size (100% of body text size) to h1 at 200%.

Now, one would think that the most natural way of doing this would be using a linear scale (100%, 120%, 140% ... 200%). However, that results in font sizes that paradoxically don't grow linearly. Here's the result of this scale, where the left image is using the default serif font, and the right image is using Palatino (introduced after #25):
<img width="250" alt="linear-serif" src="https://cloud.githubusercontent.com/assets/478237/25385353/ae42ee66-29bb-11e7-856f-4ceb6959dab2.png"> <img width="250" alt="linear-palatino" src="https://cloud.githubusercontent.com/assets/478237/25385357/b2654fe8-29bb-11e7-880a-f7d71717906e.png">

Indeed, if we think some more about this, we realize that such a scaling method is not even mathematically sound (as in describing a pattern that can be cleanly generalized), since it wouldn't work well if we were to extend it to even smaller heading sizes: we'd quickly reach size zero at h12, and beyond that we'd have *negative* font sizes!

-----

A more elegant solution would then be using an exponential function to fit the two extremes (100% at the lower end, and 200% at the upper end), and of course, the natural exponential function is the perfect candidate.

We already know that `exp(0) = 1`, so that can be our lower limit. We now want `exp(?) = 2`, which (as [brilliantly elucidated](https://www.youtube.com/watch?v=sULa9Lc4pck) by 3Blue1Brown), we can solve by `? = ln(2) ≈ 0.693`.

Now we just need to go [from 0 to 0.693 in steps of 1/5 that length](https://en.wikipedia.org/wiki/Off-by-one_error#Fencepost_error) and take the value of the exponential function at each of those points. WolframAlpha helpfully [points out](http://www.wolframalpha.com/input/?i=Table%5Bexp(ln(2)%C3%97n%2F5),+%7Bn,+0,+5%7D%5D) that the result can be simplified to the neat sequence `2^(0/5), 2^(1/5), 2^(2/5), 2^(3/5), 2^(4/5), 2^(5/5)`, thus yielding the final result:

```css
h1 { font-size: 200.00%; /* 2^(5/5) = 2^1 = 2 */ }
h2 { font-size: 174.11%; /* 2^(4/5) */ }
h3 { font-size: 151.57%; /* 2^(3/5) */ }
h4 { font-size: 131.95%; /* 2^(2/5) */ }
h5 { font-size: 114.87%; /* 2^(1/5) */ }
h6 { font-size: 100.00%; /* 2^(0/5) = 2^0 = 1 */ }
```

And indeed, behold the final rendered page (again, default serif font on the left, Palatino on the right):

<img width="250" alt="exp-serif" src="https://cloud.githubusercontent.com/assets/478237/25386781/6c4b4fbc-29c0-11e7-8927-5e9decfa5302.png"> <img width="250" alt="exp-palatino" src="https://cloud.githubusercontent.com/assets/478237/25386778/69f8a516-29c0-11e7-9246-8d0131b940ab.png">

(It would be nice to use `calc()` to provide precise values rather than approximations, but alas, only [elementary arithmetic](https://en.wikipedia.org/wiki/Elementary_arithmetic) operators are [available](https://developer.mozilla.org/en/docs/Web/CSS/calc#Expressions), and they are [not sufficient to calculate fractional exponents](https://math.stackexchange.com/questions/21381/how-to-calculate-a-decimal-power-of-a-number).)

----

Interestingly, this scaling factor (where each step represents an increase of 2^(1/5), i.e. the 5th root of two <a href="http://www.wolframalpha.com/input/?i=2%5E(1%2F5)">≈ 1.1487...</a>) does not seem to be included by default in neither [Type-Scale.com](http://type-scale.com) nor [ModularScale.com](http://modularscale.com), but following their theme of applying musical intervals to typography, we could compare this to [five-tone equal temperament](https://en.wikipedia.org/wiki/5-TET) (just like common musical intervals refer to the 12-tone equal temperament, which divides the octave into 12 equal parts in a logarithmic scale, with a ratio equal to 2^(1/12), i.e. the 12th root of 2 ≈ 1.05946).

Owen Gregory's "[Composing the New Canon: Music, Harmony, Proportion](https://24ways.org/2011/composing-the-new-canon)" has a nice table with typographic size ratios based on musical intervals of the 12-TET scale, as well as some background to the motivation for these scales. Apparently the use of 12-TET explains those tools' lack of a ratio matching the one calculated here, which, as mentioned above, uses 5-TET.

This can be easily verified in these diagrams:

![Equal temperament scales and limits](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a0/Equal_Temper_w_limits.svg/800px-Equal_Temper_w_limits.svg.png)
([source](https://commons.wikimedia.org/wiki/File:Equal_Temper_w_limits.svg))

![Equal divisions of the octave](https://upload.wikimedia.org/wikipedia/commons/6/61/Gleichstufige_Aufteilung_der_Oktave.gif)
([source](https://commons.wikimedia.org/wiki/File:Gleichstufige_Aufteilung_der_Oktave.gif))

**Comment by waldyrious** [comment]
Update: After some experimentation, I've decided to use increments in the size of the 6th root of 2 (<a href="https://www.wolframalpha.com/input/?i=2%5E%281%2F6%29">≈ 1.122462...</a>) instead, so that the smallest heading is still a bit larger than normal text size. Therefore the heading sizes will change as follows:

| Heading | Current sizes | New sizes |
|---------|---------------|-----------|
| h1 | 2<sup>5/5</sup> = 200.00% | 2<sup>6/6</sup> = 200.00% |
| h2 | 2<sup>4/5</sup> = 174.11% | 2<sup>5/6</sup> = 178.18% |
| h3 | 2<sup>3/5</sup> = 151.57% | 2<sup>4/6</sup> = 158.74% |
| h4 | 2<sup>2/5</sup> = 131.95% | 2<sup>3/6</sup> = 141.42% |
| h5 | 2<sup>1/5</sup> = 114.87% | 2<sup>2/6</sup> = 125.99% |
| h6 | 2<sup>0/5</sup> = 100.00% | 2<sup>1/6</sup> = 112.25% |
